repos/fdbt-site/cypress_tests/cypress/support/helpers.ts

```diff
@@ -344,32 +344,52 @@ export const randomlyDeterminePurchaseType = (isOtherProduct?: boolean): void =>
     continueButtonClick();
 };
 
-export const randomlyDecideTimeRestrictions = (): void => {
+export const selectTimeRestriction = () => {
+    // randomly pick a time restriction
+    getElementById('conditional-time-restriction')
+        .find('[class=govuk-radios__input]')
+        .its('length')
+        .then((length) => {
+            const randomNumber = getRandomNumber(0, length - 1);
+            getElementById('conditional-time-restriction')
+                .find('[class=govuk-radios__input]')
+                .eq(randomNumber)
+                .click()
+                .then(($radio) => {
+                    const timeRestriction = $radio.attr('value');
+                    cy.wrap(timeRestriction).as('timeRestriction');
+                });
+        });
+};
+
+export const randomlyDecideTimeRestrictions = (isEditing?: boolean): void => {
     let timeRestriction = 'N/A';
-    if (getRandomNumber(0, 1) === 0) {
-        clickElementById('valid-days-not-required');
-        cy.wrap(timeRestriction).as('timeRestriction');
-    } else {
-        // click yes button
+    const randomNumber = getRandomNumber(0, 1);
+    cy.wrap(timeRestriction).as('timeRestriction');
+    let noSelected = true;
+    cy.wrap(noSelected).as('noSelected');
+    if (isEditing) {
+        cy.get('.govuk-radios__input').each(($radio) => {
+            if ($radio.attr('value') === 'no' && $radio.prop('checked')) {
+                noSelected = true;
+                cy.wrap(noSelected).as('noSelected');
+            }
+        });
+        if (!noSelected) {
+            if (randomNumber === 0) {
+                // if yes is selected and the random number says to select no
+                clickElementById('valid-days-not-required');
+            } else {
+                // if yes is selected and the random number says to select yes change time restriction
+                selectTimeRestriction();
+            }
+        }
+    }
+    if (randomNumber === 1 && noSelected) {
```

Comment:
This block of code can run after the 'isEditing' if block has run - is that intended? Or is there a reason why this block of code cannot run in addition?

Reply:
so when creating a product you also want to make a random choice whether you make a time restriction or not. When editing if the no radio is default and the random number says to select 1 then you can select a time restriction but you have to click the yes radio because that was no selected before